src/elements/index.ts

```diff
@@ -0,0 +1,158 @@
+  const componentNames = options.components.split(',');
+  for (let component of componentNames) {
+    // using short name ("menu" for a component named "MenuComponent")
+    // convert to fully best practice name
+    const isShortName = component.toLowerCase().indexOf('component') === -1;
+    let selector = `${workspacePrefix ? `${workspacePrefix}-` : ''}`;
+    if (isShortName) {
+      selector += component.toLowerCase();
+    } else {
+      const parts = component.toLowerCase().split('component');
+      selector += parts[0];
+    }
+    componentSymbols.push({
+      selector,
+      symbol: `${stringUtils.classify(component)}${isShortName ? 'Component' : ''}`
+    });
+  }
+  componentSymbolList = componentSymbols.map(c => c.symbol).join(', ');
+
+  customElementList = createCustomElementList(componentSymbols);
+
+  return chain([
+    prerun(options),
+    // add custom element module
+    (tree: Tree) => addFiles(options),
+    // (tree: Tree, context: SchematicContext) =>
+    //   options.sample || options.routing
+    //   ? addAppFiles(options, options.sample ? "sample" : "routing")(tree, context)
+    //   : noop()(tree, context),
+    // adjust app files
+    // (tree: Tree) => adjustAppFiles(options, tree),
+    // add build scripts
+    // (tree: Tree) => {
+    //   const platformApp = options.name.replace('-', '.');
+    //   const scripts = {};
+    //   scripts[
+    //     `clean`
+    //   ] = `npx rimraf hooks node_modules package-lock.json && npm i`;
+    //   scripts[`start.${platformApp}`] = `ng serve ${options.name}`;
+    //   return updatePackageScripts(tree, scripts);
+    // },
+    // update dependencies
+    (tree: Tree, context: SchematicContext) => {
+      return updateWorkspaceSupport(options, tree, context);
+    },
+    // formatting
+    options.skipFormat 
+      ? noop()
+      : formatFiles(options)
+  ]);
+}
+
+function addFiles(options: ElementsOptions, extra: string = ""): Rule {
+  extra = extra ? `${extra}_` : "";
+  return branchAndMerge(
+    mergeWith(
+      apply(url(`./_${extra}files`), [
+        template({
+          ...(options as any),
+          name: options.name.toLowerCase(),
+          customElementList,
+          componentSymbolList,
+          componentSymbols,
+          npmScope: getNpmScope(),
+          prefix: getPrefix(),
+          dot: ".",
+          utils: stringUtils
+        }),
+        move(`xplat/web/elements`)
+      ])
+    )
+  );
+}
+
+function updateWorkspaceSupport(options: ElementsOptions, tree: Tree, context: SchematicContext) {
+  return updateJsonInTree("package.json", json => {
+    json.scripts = json.scripts || {};
+    json.dependencies = json.dependencies || {};
+    const angularVersion = json.dependencies['@angular/core'];
+    json.dependencies = {
+      ...json.dependencies,
+      "@angular/elements": angularVersion,
+      "@webcomponents/webcomponentsjs": "^2.2.7",
+      "http-server": "^0.11.1",
```

Comment:
We shouldn't need http server unless there are providing a testing index.html page ground for them, to load the exported elements in. That does seem worth while though, maybe as a e2e test or something in a ci/cd process.

Reply:
I'll be adding ability to quick preview your elements after you generate them to verify they work with this 👍